src/DynamicExpresso.Core/Parsing/Parser.cs

```diff
@@ -499,6 +499,41 @@ private Expression ParseAdditive()
 			return left;
 		}
 
+		// << , >> operators
+		private Expression ParseShift()
+		{
+			var left = ParseAdditive();
+			while (IsShiftOperator(out var shiftType))
+			{
+				NextToken();
+				var right = ParseAdditive();
+				CheckAndPromoteOperands(typeof(ParseSignatures.IShiftSignatures), ref left, ref right);
+				left = GenerateBinary(shiftType, left, right);
+			}
+			return left;
+		}
+
+		public bool IsShiftOperator(out ExpressionType shiftType)
+		{
+			// >> is not a token, because it conflicts with generics such as List<List<int>>
+			if (_token.id == TokenId.GreaterThan && _parseChar == '>')
+			{
+				NextToken(); // consume next >
+				shiftType = ExpressionType.RightShift;
+				return true;
+			}
+			// << could be a token, but is not for symmetry
+			else if (_token.id == TokenId.LessThan && _parseChar == '<')
+			{
+				NextToken(); // consume next < 
+				shiftType = ExpressionType.LeftShift;
+				return true;
+			}
+
+			shiftType = ExpressionType.DebugInfo; // dummy expression type
```

Comment:
Maybe int shi case we can set it to `null`? I don't like using null but in this case it should be safe and more clear I think.

Reply:
`shiftType` is an enum, so it can't be assigned to null. I could change the signature to use a nullable enum value, but that would require the caller to call the `Value` property, which doesn't seem ideal. In any case, if the method returns false, the `out` value should not be used. 

I'll add more comments to explain why it's not set to null.

Reply:
Ah yes, sorry. You are right. Ok for me.